app/src/test/webapp/app/shared/esbmsgsvc.test.js

```diff
@@ -0,0 +1,37 @@
+describe("EsbMessageService", function() {
+  var EsbMessageService;
+  var $httpBackend;
+
+  var resubmitRequest = {
+    "id": 1,
+    "payload": {
+      "body": "<foo>bar</foo>"
+    }
+  };
+
+  var resubmitResponse = {
+    "status": "Status"
+  };
+
+  beforeEach(module("esbMessageAdminApp"));
+
+  beforeEach(inject(function(_$httpBackend_, _EsbMessageService_){
+    $httpBackend = _$httpBackend_;
+    EsbMessageService = _EsbMessageService_;
+  }));
+
+  describe("after calling resubmit", function(){
+
+    it("should POST resubmitted payload to api/key/resubmit/{id} and return the response", function() {
+      $httpBackend.expectPOST("api/key/resubmit/1", resubmitRequest.payload).respond(200, resubmitResponse);
+      EsbMessageService.resubmitMessage(resubmitRequest)
+        .then(function(response) {
+        expect(response.data).toEqual(resubmitResponse);}
```

Comment:
(In other words, there could be a bug in `resubmitMessage` where the promise does not resolve, and the test would still pass, because the only way it could fail is if that promise resolves. By using a `done()` callback, we ensure that we know the promise has resolved and the expectation is met. Otherwise we risk never calling `expect` and the test passing erroneously.)